sdk/trace/batch_span_processor.go

```diff
@@ -120,6 +139,55 @@ func NewBatchSpanProcessor(exporter SpanExporter, options ...BatchSpanProcessorO
 	return bsp
 }
 
+var processorID atomic.Int64
```

Comment:
nit:

maybe lets name it [sequence](https://www.geeksforgeeks.org/sql/sql-sequences/) 

```suggestion
var processorIDSeq atomic.Int64
```

or counter

```suggestion
var processorIDCnt atomic.Int64
```

Reply:
used processorIDCounter to match the other PR